src/net/sourceforge/kolmafia/textui/langserver/FilesMonitor.java

```diff
@@ -0,0 +1,103 @@
+package net.sourceforge.kolmafia.textui.langserver;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import net.java.dev.spellcast.utilities.DataUtilities;
+import net.sourceforge.kolmafia.KoLConstants;
+
+/**
+ * Module in charge of the operations regarding our files, {@link Script scripts} and {@link
+ * Script.Handler handlers}, such as finding them, making them or traversing them.
+ */
+public class FilesMonitor {
+  final AshLanguageServer parent;
+
+  FilesMonitor(final AshLanguageServer parent) {
+    this.parent = parent;
+  }
+
+  /** Fetches or makes a Script for the given file. */
+  private Script getScript(final File file) {
+    synchronized (this.parent.scripts) {
+      Script script = this.parent.scripts.get(file);
+      if (script == null) {
+        script = new Script(this.parent, file);
+        this.parent.scripts.put(file, script);
+      }
+
+      return script;
+    }
+  }
+
+  /** Returns the currently existing {@link Script.Handler handler(s)} for the given file. */
+  List<Script.Handler> findHandlers(final File file) {
```

Comment:
`public`, given `findOrMakeHandlers` is public?

Reply:
Script.Handler has a reason to look for the existence of a Handler for a given File (to see if they have been replaced), but not the rest. They shouldn't have access to this method.

Reply:
(FilesMonitor was originally in the same file as Script, but was separated due to it getting somewhat bulky. Otherwise the method would have been private)

Reply:
re: the below discussion, can you make this non-package local?

Reply:
but doesn't specifically fall in the "good reason to have something be package local" that you listed? (namely, the second one)

> Good reasons I'm aware of include:
> 
>     wanting a field / method to be accessible in a test (accompanied with a comment saying it should be private)
>     splitting a series of former nested classes into their own namespace, and not wanting them to be generally public

Reply:
It does, but later on Veracity posted:
> I completely agree with wanting explicit private, protected, or public for methods and fields.

We're both maintainers. If I say "Can you do X, unless Y?" and Veracity says "Please do X", the result is that you should do X, even if Y.